hub/auth/secretcache.go

```diff
@@ -0,0 +1,173 @@
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"sync"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// SecretStoreInterface defines the interface for secret key storage
+type SecretStoreInterface interface {
+	AddSecretKey(secretKey *SecretKey, username, justification string) error
+	GetSecretKeyByOrigin(origin string) (*SecretKey, error)
+	GetDefaultSecretKey() (*SecretKey, error)
+	GetAllSecretKeys() ([]*SecretKey, error)
+	GetSecretKeyByID(id primitive.ObjectID) (*SecretKey, error)
+	UpdateSecretKey(secretKey *SecretKey, username, justification string) error
+	DisableSecretKey(id primitive.ObjectID, username, justification string) error
+}
+
+// SecretCache provides an in-memory cache for secret keys
+type SecretCache struct {
+	store        SecretStoreInterface
+	keys         map[string][]byte // Map of origin to key
+	defaultKey   []byte            // Default key
+	mutex        sync.RWMutex      // To protect concurrent access
+	lastRefresh  time.Time         // Time of last cache refresh
+	refreshEvery time.Duration     // How often to refresh the cache
+}
+
+// NewSecretCache creates a new secret key cache
+func NewSecretCache(store SecretStoreInterface, refreshInterval time.Duration) *SecretCache {
+	cache := &SecretCache{
+		store:        store,
+		keys:         make(map[string][]byte),
+		refreshEvery: refreshInterval,
+	}
+
+	// Initialize the cache
+	cache.Refresh()
+
+	return cache
+}
+
+// Refresh updates the cache with the latest keys from the database
+func (c *SecretCache) Refresh() error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	// Get all active secret keys
+	secretKeys, err := c.store.GetAllSecretKeys()
+	if err != nil {
+		return err
+	}
+
+	// If there are no keys at all, create a default one
+	if len(secretKeys) == 0 {
+		// Generate a random key for security
+		randomKey, err := generateRandomKey(32) // 32 bytes for good security
+		if err != nil {
+			return err
+		}
+
+		// Create a new default key
+		defaultKey := &SecretKey{
```

Comment:
We are not supplying `user_identifier_claim` here for the default key, but when editing it via the hub the User Identifier Claim is mandatory and has to be provided - should that be expected? Maybe some default claim should be applied here as well?

Reply:
Fixed in [commit](https://github.com/shamanec/GADS/pull/195/commits/0fb01fcb89f9ee4e0d808776e6449fbd85e646c8)